cluster-autoscaler/cloudprovider/azure/azure_vms_pool.go

```diff
@@ -0,0 +1,174 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2022-08-01/compute"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/autoscaler/cluster-autoscaler/cloudprovider"
+	"k8s.io/autoscaler/cluster-autoscaler/config"
+	"k8s.io/autoscaler/cluster-autoscaler/config/dynamic"
+	schedulerframework "k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+// VMsPool is single instance VM pool
+// this is a placeholder for now, no real implementation
+type VMsPool struct {
+	azureRef
+	manager       *AzureManager
+	resourceGroup string
+
+	minSize int
+	maxSize int
+
+	curSize int64
+	// sizeMutex       sync.Mutex
+	// lastSizeRefresh time.Time
+}
+
+// NewVMsPool creates a new VMsPool
+func NewVMsPool(spec *dynamic.NodeGroupSpec, am *AzureManager) *VMsPool {
+	nodepool := &VMsPool{
+		azureRef: azureRef{
+			Name: spec.Name,
+		},
+
+		manager:       am,
+		resourceGroup: am.config.ResourceGroup,
+
+		curSize: -1,
+		minSize: spec.MinSize,
+		maxSize: spec.MaxSize,
+	}
+
+	return nodepool
+}
+
+// MinSize returns the minimum size the cluster is allowed to scaled down
+// to as provided by the node spec in --node parameter.
+func (agentPool *VMsPool) MinSize() int {
+	return agentPool.minSize
+}
+
+// Exist is always true since we are initialized with an existing agentpool
+func (agentPool *VMsPool) Exist() bool {
+	return true
+}
+
+// Create creates the node group on the cloud provider side.
+func (agentPool *VMsPool) Create() (cloudprovider.NodeGroup, error) {
+	return nil, cloudprovider.ErrAlreadyExist
+}
+
+// Delete deletes the node group on the cloud provider side.
+func (agentPool *VMsPool) Delete() error {
+	return cloudprovider.ErrNotImplemented
+}
+
+// Autoprovisioned is always false since we are initialized with an existing agentpool
+func (agentPool *VMsPool) Autoprovisioned() bool {
+	return false
+}
+
+// GetOptions returns NodeGroupAutoscalingOptions that should be used for this particular
+// NodeGroup. Returning a nil will result in using default options.
+func (agentPool *VMsPool) GetOptions(defaults config.NodeGroupAutoscalingOptions) (*config.NodeGroupAutoscalingOptions, error) {
+	// TODO(wenxuan): Implement this method
+	return nil, cloudprovider.ErrNotImplemented
+}
+
+// MaxSize returns the maximum size scale limit provided by --node
+// parameter to the autoscaler main
+func (agentPool *VMsPool) MaxSize() int {
+	return agentPool.maxSize
+}
+
+// TargetSize returns the current TARGET size of the node group. It is possible that the
+// number is different from the number of nodes registered in Kubernetes.
+func (agentPool *VMsPool) TargetSize() (int, error) {
+	// TODO(wenxuan): Implement this method
+	return -1, cloudprovider.ErrNotImplemented
```

Comment:
I am assuming we'll be implementing all of these in follow-up PRs?

Reply:
Yes that's right, to fully implement all the methods we will need the agentpool client PR to be merged first.

I'm trying to keep the PR size to be small and self-contained, so it's easier to review. There will be more PRs coming for the implementations.